spec/src/main/asciidoc/architecture.adoc

```diff
@@ -182,15 +182,19 @@ to `true` and have multiple methods annotated as metric that have the same metho
 .Example of reused counters
 [source,java]
 ----
-    @Counted(name = "countMe", absolute = true, reusable = true)
+    @HitCounted(name = "countMe", absolute = true, reusable = true)
     public void countMeA() { }
 
-    @Counted(name = "countMe", absolute = true, reusable = true)
+    @HitCounted(name = "countMe", absolute = true, reusable = true)
     public void countMeB() { }
 ----
 
 In the above examples both `countMeA()` and `countMeB()` will share a single Counter with registered name `countMe` in application scope.
 
+The implementation must throw an 'IllegalArgumentException' if a counter defined via `@HitCounted` is reused as `@ParallelCounted` or vice versa.
```

Comment:
Actually the more I think about it, the more I think we should allow this. Imagine a JAX-RS endpoint, where the HitCount counts total invocations and the parallelcount counts the parallel ones. 
But then from a pure "expose metrics to a 3rd party" point of view, parallelcounted is probably not very useful if we don't keep the max around until it is retrieved (and then reset back to 0).